lib/spack/spack/cmd/module.py

```diff
@@ -208,8 +213,11 @@ def find(module_types, specs, args):
         msg += 'no {0} module has been generated for it.'
         tty.die(msg.format(module_type, spec))
 
-    # ... and if it is print its use name
-    print(writer.layout.use_name)
+    # ... and if it is print its use name or full-path if requested
+    if args.full_path:
+        print(writer.layout.filename)
```

Comment:
To add a test case for this, you just need to change:
```python
def test_find(database, parser):
    """Tests the 'spack module find' under a few common scenarios."""

    # Try to find it for tcl module files
    args = parser.parse_args(['find', '--module-type', 'tcl', 'libelf'])
    module.module(parser, args)
```
in `lib/spack/spack/test/cmd/module.py` with:
```python
@pytest.mark.parametrize('cli_args', [
    ['--module-type', 'tcl', 'libelf'],
    ['--module-type', '--full-path', 'tcl', 'libelf']
])
def test_find(database, parser, cli_args):
    """Tests the 'spack module find' under a few common scenarios."""

    # Try to find it for tcl module files
    args = parser.parse_args(['find'] + cli_args)
    module.module(parser, args)
```

Reply:
Added the test.

Our use case: we are building packages centrally on a build server, and then syncing to a rsync/nfs round-robin pair for production use around our University. Having this option lets our sync script easily find the location for the lmod file to be synced (using spack location -i for the app)